### PR TITLE
Homework3: Alexey Masharov

### PR DESCRIPTION
**Pull request by JIekceu**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/Homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+кратных 3 пишет Fizz, вместо чисел кратный 5 пишет Buzz,
+а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz
+'''
+for i in range(1, 101):
+    prn = i
+    if i % 3 == 0:
+        prn = 'Fizz'
+    if i % 5 == 0:
+        prn = 'Buzz'
+    if i % 3 == 0 and i % 5 == 0:
```
Comment (belovmd) [inline src/Homework3/task1.py]: 15?

File: src/Homework3/task2.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+2.	Используйте на предыдущий список slice чтобы получить следующий: ['ab', 'ad', 'bc'].
+3.	Используйте генератор списков чтобы получить следующий ['1a', '2a', '3a', '4a'].
+4.	Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5.	Скопируйте список и добавьте в него элемент '2a' так,
+    чтобы в исходном списке этого элемента не было.
+Задачу поместите в файл task2.py в папке src/homework3.
+
+'''
+import copy
+list1 = [el1 + el2 for el1 in ['a', 'b'] for el2 in ['b', 'c', 'd']]
```
Comment (belovmd) [inline src/Homework3/task2.py]: 'ab'

File: src/Homework3/task2.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+print('1. ', list1)
+list2 = list1[::2]
+print('2. ', list2)
+list3 = [str(el) + 'a' for el in range(1, 5)]
```
Comment (belovmd) [inline src/Homework3/task2.py]: '1234'?

File: src/Homework3/task4.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+counter_par = int()
+antirepeat = []
+str_ = '1 1 1 2 2 2 2 3 3 3 3 3'
+list_ = [int(elem) for elem in str_.split()]
+for element in list_:
+    if element not in antirepeat:
+        counter_elements = list_.count(element)
+        antirepeat.append(element)
+        if counter_elements > 1:
+            for factorial in range(1, counter_elements):  # counter_par = (counter_elements - 1)!
```
Comment (belovmd) [inline src/Homework3/task4.py:19]: has O(1) solution

**Review by belovmd — CHANGES_REQUESTED** [review]

File: src/Homework3/task1.py (hunk 1 above)
Comment (JIekceu) [reply src/Homework3/task1.py]: fixed

File: src/Homework3/task4.py (hunk 4 above)
Comment (JIekceu) [reply src/Homework3/task4.py:19]: it isn't second solution, it's just a comment

**Comment by belovmd** [comment]
9

**Pull request closed without merging**